File: 18_xau_nhi_phan_kbit_1.py
def inxau():
    global n,M
    for i in range(0,n):
        print(M[i],end='')
    print()

def kbit1():
    global n , k ,M, cauhinhcuoi
    i=n-1
    while M[i]==1 and i>=0:
        M[i]= 0 
        i-=1
    if i==-1:
        cauhinhcuoi = True
        return
    M[i] = 1 
    count = 0 
    for i in range(0 , n):
        if M[i]==1:
            count+=1
    if count == k:
        inxau()
        
t = int(input())
while t>0:
    n , k = map(int , input().split())
    cauhinhcuoi = False
    M = []
    for i in range(1 , n+1):
        M.append(0) 
    while cauhinhcuoi == False:
        kbit1()
    t-=1

File: 18_xau_nhi_phan_kbit_1.cpp
#include<iostream>
using namespace std ; 
int n , k ; int M[17] ; 
bool caucauhinhcuoi ; 
void in() {
    for (int i=1 ; i<=n ; i++) {
        cout<<M[i];
    }
    cout<<endl ;
}
void kbit1() {
    int i=n ; 
    while(M[i]==1 and i>=1) {
        M[i] = 0 ; 
        i-- ; 
    }
    if (i==0) {
        caucauhinhcuoi = true ; 
        return  ;
    }
    M[i] = 1;
    int count=0 ; 
    for(int i=1 ; i<=n ; i++) {
        if(M[i]==1) count++ ; 
    } 
    if(count == k) in() ;
}
main(){
    int t  ; cin>>t; 
    while(t--){
        cin>>n>>k;
        for(int i=1 ;i<=n ; i++) M[i]==0 ;
        caucauhinhcuoi = false ; 
        while(caucauhinhcuoi == false ) {
            kbit1() ; 
        }
    }
}